File: server/src/models/Onboarding.ts
import { DataTypes } from 'sequelize';
import sequelize from '../config/database';

const Onboarding = sequelize.define('Onboarding', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true
  },
  fullName: {
    type: DataTypes.STRING(255),
    allowNull: false,
    field: 'full_name'
  },
  email: {
    type: DataTypes.STRING(255),
    allowNull: false,
    unique: true,
    validate: {
      isEmail: true
    }
  },
  university: {
    type: DataTypes.STRING(255),
    allowNull: false
  },
  degreeType: {
    type: DataTypes.ENUM('Bachelor\'s', 'Master\'s', 'PhD'),
    allowNull: false,
    field: 'degree_type'
  },
  major: {
    type: DataTypes.STRING(255),
    allowNull: false
  },
  enrollmentYear: {
    type: DataTypes.STRING(10),
    allowNull: false,
    field: 'enrollment_year'
  },
  graduationYear: {
    type: DataTypes.STRING(10),
    allowNull: false,
    field: 'graduation_year'
  },
  thesisTitle: {
    type: DataTypes.TEXT,
    allowNull: true,
    field: 'thesis_title'
  },
  supervisorName: {
    type: DataTypes.STRING(255),
    allowNull: true,
    field: 'supervisor_name'
  },
  thesisStage: {
    type: DataTypes.STRING(100),
    allowNull: true,
    field: 'thesis_stage'
  },
  researchInterests: {
    type: DataTypes.TEXT,
    allowNull: true,
    field: 'research_interests'
  },
  familiarTools: {
    type: DataTypes.TEXT,
    allowNull: true,
    field: 'familiar_tools'
  },
  helpNeeded: {
    type: DataTypes.TEXT,
    allowNull: true,
    field: 'help_needed'
  },
  publishPlans: {
    type: DataTypes.BOOLEAN,
    defaultValue: false,
    field: 'publish_plans'
  },
  hasDocuments: {
    type: DataTypes.BOOLEAN,
    defaultValue: false,
    field: 'has_documents'
  },
  documentPaths: {
    type: DataTypes.TEXT,
    allowNull: true,
    field: 'document_paths'
  },
  onboardingCompleted: {
    type: DataTypes.BOOLEAN,
    defaultValue: false,
    field: 'onboarding_completed'
  }
}, {
  tableName: 'user_onboarding',
  timestamps: true,
  createdAt: 'created_at',
  updatedAt: 'updated_at'
});

export default Onboarding;

File: backend/models/onboarding.py
from sqlalchemy import Column, String, DateTime, Text, Boolean
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.sql import func
from core.database import Base
import uuid

class UserOnboarding(Base):
    __tablename__ = "user_onboarding"
    
    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    full_name = Column(String(255), nullable=False)
    email = Column(String(255), nullable=False, unique=True)
    university = Column(String(255), nullable=False)
    degree_type = Column(String(50), nullable=False)  # Bachelor's, Master's, PhD
    major = Column(String(255), nullable=False)
    enrollment_year = Column(String(10), nullable=False)
    graduation_year = Column(String(10), nullable=False)
    thesis_title = Column(Text, nullable=True)
    supervisor_name = Column(String(255), nullable=True)
    thesis_stage = Column(String(100), nullable=True)  # Planning, Literature Review, etc.
    research_interests = Column(Text, nullable=True)
    familiar_tools = Column(Text, nullable=True)  # JSON array of tools
    help_needed = Column(Text, nullable=True)  # JSON array of help types
    publish_plans = Column(Boolean, default=False)
    has_documents = Column(Boolean, default=False)
    document_paths = Column(Text, nullable=True)  # JSON array of document paths
    onboarding_completed = Column(Boolean, default=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
